packages/plugin-essentials/sources/commands/explain/peerRequirements.ts

```diff
@@ -0,0 +1,180 @@
+    configuration,
+    stdout: opts.stdout,
+    includeFooter: false,
+  }, async report => {
+    const subject = project.storedPackages.get(requirement.subject);
+    if (typeof subject === `undefined`)
+      throw new Error(`Assertion failed: Expected the subject package to have been registered`);
+
+    const rootRequester = project.storedPackages.get(requirement.rootRequester);
+    if (typeof rootRequester === `undefined`)
+      throw new Error(`Assertion failed: Expected the root package to have been registered`);
+
+    const providedDescriptor = subject.dependencies.get(requirement.requested.identHash) ?? null;
+
+    const providedResolution = providedDescriptor !== null
+      ? project.storedResolutions.get(providedDescriptor.descriptorHash)
+      : null;
+
+    if (typeof providedResolution === `undefined`)
+      throw new Error(`Assertion failed: Expected the resolution to have been registered`);
+
+    const provided = providedResolution !== null
+      ? project.storedPackages.get(providedResolution)
+      : null;
+
+    if (typeof provided === `undefined`)
+      throw new Error(`Assertion failed: Expected the provided package to have been registered`);
+
+    const allRequesters = [...requirement.allRequesters.values()].map(requesterHash => {
+      const pkg = project.storedPackages.get(requesterHash);
+      if (typeof pkg === `undefined`)
+        throw new Error(`Assertion failed: Expected the package to be registered`);
+
+      const devirtualizedLocator = structUtils.devirtualizeLocator(pkg);
+      const devirtualizedPkg = project.storedPackages.get(devirtualizedLocator.locatorHash);
+      if (typeof devirtualizedPkg === `undefined`)
+        throw new Error(`Assertion failed: Expected the package to be registered`);
+
+      const peerDependency = devirtualizedPkg.peerDependencies.get(requirement.requested.identHash);
+      if (typeof peerDependency === `undefined`)
+        throw new Error(`Assertion failed: Expected the peer dependency to be registered`);
+
+      return {pkg, peerDependency};
+    });
+
+    if (provided !== null) {
+      const satisfiesAllRanges = allRequesters.every(({peerDependency}) => {
+        return semverUtils.satisfiesWithPrereleases(provided.version, peerDependency.range);
+      });
+
+      report.reportInfo(MessageName.UNNAMED, `${
+        structUtils.prettyLocator(configuration, subject)
+      } provides ${
+        structUtils.prettyLocator(configuration, provided)
+      } with version ${
+        structUtils.prettyReference(configuration, provided.version ?? `<missing>`)
+      } which ${satisfiesAllRanges ? `satisfies` : `doesn't satisfy`} the following requirements:`);
+    } else {
+      report.reportInfo(MessageName.UNNAMED, `${
+        structUtils.prettyLocator(configuration, subject)
+      } doesn't provide ${
+        structUtils.prettyIdent(configuration, requirement.requested)
+      }, breaking the following requirements:`);
+    }
```

Comment:
What's the reason for not showing the root requester? It's currently printed both inside the warnings and on `yarn explain peer-requirements` without a hash, but not on `yarn explain peer-requirements <hash>`. Since it's quite useful to know which one of the requesters is the root requester and because it doesn't take much space to print it, I think we should. What do you think?

Reply:
iirc it was mostly line length, and it was duplicate with the entry from the list (although it's true it doesn't show at all what's the root anymore). For now I think it should be fine; to see this list you have to go through `yarn explain peer-dependencies` or a warning, both of which show the root.

Reply:
🤔 I still feel like showing the root (which would only come at the cost of a few extra characters) would be a good idea, as it's one of the 3 things that describe a set of peer requirements (the parent, the root, and the peer request). Omitting it still feels wrong to me.